config/routes.rb

```diff
@@ -8,4 +8,5 @@
   end
   get "/auth/github", as: :github_login
   get "/auth/github/callback", to: "sessions#create"
+  get "/start_auth", to: 'github#github_start'
```

Comment:
I agree, Im a bit confused as to the functionality of the session_id and user_name routes.